Mathlib/LinearAlgebra/LinearIndependent.lean

```diff
@@ -275,6 +275,22 @@ theorem LinearIndependent.map' (hv : LinearIndependent R v) (f : M →ₗ[R] M')
   hv.map <| by simp [hf_inj]
 #align linear_independent.map' LinearIndependent.map'
 
+/-- If `M / R` and `M' / R'` are modules, `i : R → R'` is a surjective map which maps zero to zero,
+`j : M →+ M'` is an injective monoid map, such that the scalar multiplications on `M` and `M'` are
+compatible, then `j` sends linearly independent families of vectors to linearly independent
+families of vectors. As a special case, taking `R = R'` it is `LinearIndependent.map'`. -/
+theorem LinearIndependent.map_of_surjective_injective {R' : Type*} {M' : Type*}
+    [Semiring R'] [AddCommMonoid M'] [Module R' M'] (hv : LinearIndependent R v)
+    (i : ZeroHom R R') (j : M →+ M') (hi : Surjective i) (hj : Injective j)
+    (hc : ∀ (s : R) (x : M), j (s • x) = i s • j x) : LinearIndependent R' (j ∘ v) := by
```

Comment:
I know this condition is similar to [SemilinearMapClass](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Algebra/Module/LinearMap.html#SemilinearMapClass), but it seems the more morally correct statement is `i : ZeroHom R' R`, `Injective i` and `j (i r' • m) = r' • j m` for `r' : R` and `m : M` in this situation. And it seems strictly more general.

Reply:
Thanks. Do you have any suggestions on the `[lift]_rank_eq` version? Currently I keep the condition of them unchanged.

Reply:
The eq versions uses bijections so I suppose it doesn't matter which direction they're in?